### ISSUE TITLE
Vulnerabilities in several dependencies

### ISSUE THREAD

**Issue by berkeli** [kubernetes-sigs/aws-load-balancer-controller#3217]
Hello, 

We noticed there are several vulnerabilities in dependencies as listed below:

```
Scan results for: image public.ecr.aws/eks/aws-load-balancer-controller:v2.5.2 sha256:56c320e95b5e664743777b3e75b52c95a1e77893e25767fe8f7c139148636389
Vulnerabilities
+------------------+----------+------+-----------------------------------+---------+-------------------------+-------------+------------+----------------------------------------------------+
|       CVE        | SEVERITY | CVSS |              PACKAGE              | VERSION |         STATUS          |  PUBLISHED  | DISCOVERED |                    DESCRIPTION                     |
+------------------+----------+------+-----------------------------------+---------+-------------------------+-------------+------------+----------------------------------------------------+
| CVE-2023-24540   | critical | 9.80 | go                                | 1.20.3  | fixed in 1.20.4, 1.19.9 | 19 days     | < 1 hour   | Not all valid JavaScript whitespace characters are |
|                  |          |      |                                   |         | 7 days ago              |             |            | considered to be whitespace. Templates containing  |
|                  |          |      |                                   |         |                         |             |            | whitespace characters outside of the character     |
|                  |          |      |                                   |         |                         |             |            | se...                                              |
+------------------+----------+------+-----------------------------------+---------+-------------------------+-------------+------------+----------------------------------------------------+
| PRISMA-2022-0227 | high     | 7.50 | github.com/emicklei/go-restful/v3 | v3.9.0  | fixed in v3.10.0        | > 11 months | < 1 hour   | github.com/emicklei/go-restful/v3 module prior     |
|                  |          |      |                                   |         | > 7 months ago          |             |            | to v3.10.0 is vulnerable to Authentication Bypass  |
|                  |          |      |                                   |         |                         |             |            | by Primary Weakness. There is an inconsistency in  |
|                  |          |      |                                   |         |                         |             |            | how...                                             |
+------------------+----------+------+-----------------------------------+---------+-------------------------+-------------+------------+----------------------------------------------------+
| CVE-2023-29400   | high     | 7.30 | go                                | 1.20.3  | fixed in 1.20.4, 1.19.9 | 19 days     | < 1 hour   | Templates containing actions in unquoted HTML      |
|                  |          |      |                                   |         | 7 days ago              |             |            | attributes (e.g. \"attr={{.}}\") executed with     |
|                  |          |      |                                   |         |                         |             |            | empty input can result in output with unexpected   |
|                  |          |      |                                   |         |                         |             |            | results ...                                        |
+------------------+----------+------+-----------------------------------+---------+-------------------------+-------------+------------+----------------------------------------------------+
| CVE-2023-24539   | high     | 7.30 | go                                | 1.20.3  | fixed in 1.20.4, 1.19.9 | 19 days     | < 1 hour   | Angle brackets (<>) are not considered dangerous   |
|                  |          |      |                                   |         | 7 days ago              |             |            | characters when inserted into CSS contexts.        |
|                  |          |      |                                   |         |                         |             |            | Templates containing multiple actions separated by |
|                  |          |      |                                   |         |                         |             |            | a \'/\...                                          |
+------------------+----------+------+-----------------------------------+---------+-------------------------+-------------+------------+----------------------------------------------------+

Vulnerabilities found for image public.ecr.aws/eks/aws-load-balancer-controller:v2.5.2: total - 4, critical - 1, high - 3, medium - 0, low - 0
```

Just wanted to bring these to your attention as I didn't see any issues raised yet. Happy to take a look at them myself too if there's no ongoing work to fix these. 

Thanks!

**Comment by oliviassss** [comment]
@berkeli Thanks for the info, may I know which tool you used for scanning the image?

**Comment by berkeli** [comment]
The scan is done with https://www.paloaltonetworks.com/prisma/cloud

**Comment by oliviassss** [comment]
@berkeli, we have addressed the CVEs in our patch release v2.5.3, I'm closing this issue for now. Feel free to reach out or reopen if there's any issue. Thanks.

**Comment by berkeli** [comment]
Thanks @oliviassss for the quick turnaround. 

It has resolved the vulnerabilities I listed originally, but there seem to be few remaining (fairly new).

```
Scan results for: image public.ecr.aws/eks/aws-load-balancer-controller:v2.5.3 sha256:674f1ef419ab2ac0026ee6fff4262ab6007a93ac9dae72e94aa585031fac7fe6
Vulnerabilities
+----------------+----------+------+---------+---------+--------------------------+-----------+------------+----------------------------------------------------+
|      CVE       | SEVERITY | CVSS | PACKAGE | VERSION |          STATUS          | PUBLISHED | DISCOVERED |                    DESCRIPTION                     |
+----------------+----------+------+---------+---------+--------------------------+-----------+------------+----------------------------------------------------+
| CVE-2023-29405 | critical | 9.80 | go      | 1.20.3  | fixed in 1.20.5, 1.19.10 | 18 days   | < 1 hour   | The go command may execute arbitrary code at build |
|                |          |      |         |         | 11 days ago              |           |            | time when using cgo. This may occur when running   |
|                |          |      |         |         |                          |           |            | \"go get\" on a malicious module, or when running  |
|                |          |      |         |         |                          |           |            | ...                                                |
+----------------+----------+------+---------+---------+--------------------------+-----------+------------+----------------------------------------------------+
| CVE-2023-29404 | critical | 9.80 | go      | 1.20.3  | fixed in 1.20.5, 1.19.10 | 18 days   | < 1 hour   | The go command may execute arbitrary code at build |
|                |          |      |         |         | 11 days ago              |           |            | time when using cgo. This may occur when running   |
|                |          |      |         |         |                          |           |            | \"go get\" on a malicious module, or when running  |
|                |          |      |         |         |                          |           |            | ...                                                |
+----------------+----------+------+---------+---------+--------------------------+-----------+------------+----------------------------------------------------+
| CVE-2023-29402 | critical | 9.80 | go      | 1.20.3  | fixed in 1.20.5, 1.19.10 | 18 days   | < 1 hour   | The go command may generate unexpected code at     |
|                |          |      |         |         | 10 days ago              |           |            | build time when using cgo. This may result in      |
|                |          |      |         |         |                          |           |            | unexpected behavior when running a go program      |
|                |          |      |         |         |                          |           |            | which uses ...                                     |
+----------------+----------+------+---------+---------+--------------------------+-----------+------------+----------------------------------------------------+
| CVE-2023-24540 | critical | 9.80 | go      | 1.20.3  | fixed in 1.20.4, 1.19.9  | 47 days   | < 1 hour   | Not all valid JavaScript whitespace characters are |
|                |          |      |         |         | 35 days ago              |           |            | considered to be whitespace. Templates containing  |
|                |          |      |         |         |                          |           |            | whitespace characters outside of the character     |
|                |          |      |         |         |                          |           |            | se...                                              |
+----------------+----------+------+---------+---------+--------------------------+-----------+------------+----------------------------------------------------+
| CVE-2023-29403 | high     | 7.80 | go      | 1.20.3  | fixed in 1.20.5, 1.19.10 | 18 days   | < 1 hour   | On Unix platforms, the Go runtime does not         |
|                |          |      |         |         | 11 days ago              |           |            | behave differently when a binary is run with       |
|                |          |      |         |         |                          |           |            | the setuid/setgid bits. This can be dangerous in   |
|                |          |      |         |         |                          |           |            | certain cases...                                   |
+----------------+----------+------+---------+---------+--------------------------+-----------+------------+----------------------------------------------------+
| CVE-2023-29400 | high     | 7.30 | go      | 1.20.3  | fixed in 1.20.4, 1.19.9  | 47 days   | < 1 hour   | Templates containing actions in unquoted HTML      |
|                |          |      |         |         | 35 days ago              |           |            | attributes (e.g. \"attr={{.}}\") executed with     |
|                |          |      |         |         |                          |           |            | empty input can result in output with unexpected   |
|                |          |      |         |         |                          |           |            | results ...                                        |
+----------------+----------+------+---------+---------+--------------------------+-----------+------------+----------------------------------------------------+
| CVE-2023-24539 | high     | 7.30 | go      | 1.20.3  | fixed in 1.20.4, 1.19.9  | 47 days   | < 1 hour   | Angle brackets (<>) are not considered dangerous   |
|                |          |      |         |         | 35 days ago              |           |            | characters when inserted into CSS contexts.        |
|                |          |      |         |         |                          |           |            | Templates containing multiple actions separated by |
|                |          |      |         |         |                          |           |            | a \'/\...                                          |
+----------------+----------+------+---------+---------+--------------------------+-----------+------------+----------------------------------------------------+
```


Technically, these should have been resolved with https://github.com/kubernetes-sigs/aws-load-balancer-controller/pull/3253 so I'm not sure why it still thinks go version is 1.20.3? 

I have tried removing cache and pulling the image from fresh.
